src/store/fileUploadStore.js

```diff
@@ -217,13 +217,19 @@ const actions = {
 	 * @param {boolean} data.rename whether to rename the files (usually after pasting)
 	 * @param {boolean} data.isVoiceMessage whether the file is a voice recording
 	 */
-	async initialiseUpload({ commit, dispatch }, { uploadId, token, files, rename = false, isVoiceMessage }) {
+	async initialiseUpload({ commit, getters, dispatch }, { uploadId, token, files, rename = false, isVoiceMessage }) {
 		// Set last upload id
 		commit('setCurrentUploadId', uploadId)
 
+		const initialisedUploads = getters.getInitialisedUploads(uploadId).map(upload => upload[1].file)
+
 		for (let i = 0; i < files.length; i++) {
 			const file = files[i]
-
+			if (initialisedUploads.some(upload => upload.name === file.name
+				&& upload.size === file.size
+				&& upload.lastModified === file.lastModified)) {
+				continue
+			}
```

Comment:
Why do we need this check? Does it mean it is forbidden to send files with the same name + date + size?  (They still may have different content)

Reply:
Scenario:
1. I created an `logo.svg` and attached it
2. Then I found that I needed to move one line and change it. There is no size change. 
3. For some reason `lastModified` was the same (suddenly pruned (we have this error on Nextcloud server), or the files were generated).

Reply:
Isn't the linked issue fixed by clearing input[type=file] value?

Reply:
I noticed, that after clearing it's possible to add same file twice. Then during the upload first file will be shared normally, but second returns 423 - locked and will be shown as failed. Don't know, which way is better, that or the omitting of duplicates

Reply:
Issue is caused by this part:

https://github.com/nextcloud/spreed/blob/0727d1987da02129f66c84fa9a9eded45d398e90/src/store/fileUploadStore.js#L313-L322

It makes requests in parallel, so `findUniquePath` is called in parallel and returns the same path for 2 files that are to be uploaded together.

Reply:
It also fails if 2 files just have the same name, regardless of file size or the last modified.

Reply:
Will try to find a workaround, because parallel requests allow to save significant amount of time here, while uploading (around 1s for each file)

Reply:
We could check if we try to upload files with the same name? If we do, it will not work anyway. So we need to alter the name before checking for uniqueness with the server?

Reply:
Seems to be a problem here: when I'm trying to put the file under the same path, it got locked by Dav:
![image](https://github.com/nextcloud/spreed/assets/93392545/a48f0881-d3a9-44f6-80ad-0d6d686b5c51)

Simplest thing i could suggest is to check for duplicates before uploading:
- if there are only unique names, run PROPFIND in parallel
- if no - run consequently

Reply:
What about check for filename in JS, and if files are equal, and change the name to `{name} (index)` manually? 

There is still a risk that a user has this `{name} (index)` file already, but in this case `findUniquePath` will rename it to `{name} (index) (2)`.

Alternatively, we can add a random prefix.

Reply:
Or re-request the uniq name in catch for failed files.

Reply:
> and change the name to `{name} (index)` manually?

findUniquePath checks if path already has a suffix, removes it and only after that perform a search (although it's broken, and for me file `test (10).jpg` resulted in `test (1 (2).jpg`, and `test (6).jpg` resulted in `test ( (2).jpg`.

So, adding an upload timestamp as suffix for duplicates should be enough

Reply:
> findUniquePath checks if path already has a suffix, removes it and only after that perform a search (although it's broken, and for me file test (10).jpg resulted in test (1 (2).jpg, and test (6).jpg resulted in test ( (2).jpg.

Then the method needs fixing.

> So, adding an upload timestamp as suffix for duplicates should be enough

Not a fan of working around